### PR TITLE
support mount a bucket which already exist on S3

### PR DESCRIPTION
**Pull request by ogre0403**
Thanks for your awesome s3 csi project. However, we mount existing s3 bucket in our use case. 

In this PR, I add s3 key information in PV spec, then node server will use this information to mount bucket.   

In this PR, node server will ignore `.metadata.json` and `csi-fs` folder inside bucket, node server just mount while bucket into pod. 

Please take your time to review this PR. Thanks. 

```
apiVersion: v1
kind: PersistentVolume
metadata:
  name: new-pv
spec:
  accessModes:
    - ReadWriteOnce
  capacity:
    storage: 5Gi
  csi:
    driver: ch.ctrox.csi.s3-driver
    volumeAttributes:
      accessKeyID: <YOUR_ACCESS_KEY_ID>
      secretAccessKey: <YOUR_SECRET_ACCES_KEY>
      # For AWS set it to "https://s3.<region>.amazonaws.com"
      endpoint: https://s3.eu-central-1.amazonaws.com
      # If not on S3, set it to ""
      region: <S3_REGION>
    # Name of bucket which already created
    volumeHandle: <BUCKET_NAME>
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +85,64 @@ $ touch /var/lib/www/html/hello_world
...
+spec:
+  accessModes:
+    - ReadWriteOnce
+  capacity:
+    storage: 5Gi
+  csi:
+    driver: ch.ctrox.csi.s3-driver
+    volumeAttributes:
+      accessKeyID: <YOUR_ACCESS_KEY_ID>
+      secretAccessKey: <YOUR_SECRET_ACCES_KEY>
```
Comment (guymguym) [inline README.md]: Normally you should not have credentials in clear text here, but instead to refer from the PV to a `Secret` that will store those. This is meant to provide separate access controls over the sensitive info.

Comment (ogre0403) [reply README.md]: You are right. I'll move s3 credentials into secret, then read it from node server.

**Comment by a7i** [comment]
What are the plans for this PR? Any remaining items that I can help with?

**Comment by jcdauchy-moodys** [comment]
Is there a chance that this PR gets merged ? I guess many people would like to use this feature.

**Comment by ctrox** [comment]
Really sorry for not getting to this earlier and thanks a lot to @ogre0403 for the work here!

I have just merged https://github.com/ctrox/csi-s3/pull/42 in favour of this one, since it basically does the same thing but simplified. This PR introduces a k8s client to the driver, which is a bit of an anti-pattern. The CSI driver should not directly speak to the Kubernetes API. In fact, it should not even rely on running on Kubernetes. It is supposed to be compatible with all kinds of [container orchestrators](https://github.com/container-storage-interface/spec/blob/master/spec.md#objective).

**Pull request closed without merging**